### PR TITLE
Nadel now uses a builder pattern and allows instrumentation to be injected into it

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -72,7 +72,7 @@ You create a `Nadel` instance and execute queries on it:
 // A GraphqlCaller actually executes a request against a service
 GraphqlCallerFactory graphqlCallFactory = ....  
 String dsl = "<Nadel stitching DSL>";
-Nadel nadel = new Nadel(dsl, callerFactory);
+Nadel nadel = Nadel nadel = Nadel.newNadel().dsl(dsl).remoteRetrieverFactory(callerFactory).build()
```
Comment (johnliucn925) [inline README.MD]: duplicated definition

File: src/main/java/graphql/nadel/Nadel.java [hunk 2]
```diff
@@ -224,4 +212,54 @@ private InvalidDslException hydrationError(InnerServiceHydration hydration, Stri
...
 
+    /**
+     * @return a builder of Nadel objects
+     */
+    public static Builder newNadel() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        String dsl;
```
Comment (johnliucn925) [inline src/main/java/graphql/nadel/Nadel.java]: Do you think we could reduce these variables accessibility to private final?

Comment (bbakerman) [reply src/main/java/graphql/nadel/Nadel.java]: we can make them private but not final.  Its a Builder after all and hence mutable by design.

For the record the Builder pattern with a static `newFoo` pattern has served us really well in graphql-java.  Stops the 6 constructors pattern we had above

File: README.MD (hunk 1 above)
Comment (bbakerman) [reply README.MD]: fixed

File: src/main/java/graphql/nadel/Nadel.java (hunk 2 above)
Comment (bbakerman) [reply src/main/java/graphql/nadel/Nadel.java]: fixed

**Pull request merged**